ext/MooncakeCUDAExt.jl

```diff
@@ -33,19 +33,45 @@ import Mooncake.TestUtils:
     populate_address_map_internal, AddressMap, __increment_should_allocate
 
 const CuFloatArray = CuArray{<:IEEEFloat}
+const CuComplexArray = CuArray{<:Complex{<:IEEEFloat}}
 
 # Tell Mooncake.jl how to handle CuArrays.
 
+Mooncake.@foldable tangent_type(::Type{<:CuArray{P,N,M}}) where {P,N,M} = CuArray{
```

Comment:
As discussed in https://github.com/chalk-lab/Mooncake.jl/pull/811/files#r2451846833, let's restrict to `Union{Complex{<:IEEEFloat}, IEEEFloat}` for now and relax it when tests for new types can be added. 

```suggestion
Mooncake.@foldable tangent_type(::Type{<:CuArray{P,N,M}}) where {P<:Union{Complex{<:IEEEFloat}, IEEEFloat},N,M} = CuArray{
```